### PR TITLE
LPS-167148 Remove "immediate = true" from Echo components part 2

### PR DESCRIPTION
**Pull request by rafaprax**
Ticket: https://issues.liferay.com/browse/LPS-167148

### CODE REVIEW

**Comment by rafaprax** [comment]
Hey @shuyangzhou,

we found out why the `JournalArticleIndexer` becomes satisfied after the re-indexing process. There was an issue in [ServiceTrackerListImpl.removedService()](https://github.com/liferay-core-infra/liferay-portal/pull/2525/commits/9dcd86d8a369979415447ca07e44b4805ec921fe#diff-4c1e97d3dfbfec9f175451c2cc3828d417ccac8d35e9a8db554110e533e0d447L171-L179) that is calling the `ungetService` twice. So, as those Indexers are referenced only by the IndexerRegistryImpl and SearchEngineInitializer#_reindex, the additional call is causing the *Indexer services to deactivate and stay in the satisfied state because there are no more references for that service.

PS: We've tested with others Indexes implementations and we faced the same issue.

Thx

**Comment by rafaprax** [comment]
ci:test:echo

**Comment by rafaprax** [comment]
ci:test:startup

**Comment by zhangyan0701** [comment]
No regressions found.

**Comment by rafaprax** [comment]
Hey @shuyangzhou, this pr is ready for your review. Thx

**Comment by shuyangzhou** [comment]
@rafaprax please rebase, thanks!

**Comment by rafaprax** [comment]
ci:test:sf

**Comment by rafaprax** [comment]
ci:test:relevant

**Comment by rafaprax** [comment]
ci:test:startup

**Comment by Kvale1733** [comment]
ci:test:echo

**Comment by catarineaguiar** [comment]
@rafaprax no regressions found.

**Comment by rafaprax** [comment]
@shuyangzhou conflicts were fixed and it is ready for your review. Thx

File: modules/core/osgi-service-tracker-collections/src/main/java/com/liferay/osgi/service/tracker/collections/internal/list/ServiceTrackerListImpl.java [hunk 1]
```diff
@@ -174,8 +174,6 @@ public void removedService(
 			_serviceTrackerCustomizer.removedService(serviceReference, service);
 
 			_update(serviceReference, service, true);
-
```
Comment (tinatian) [inline modules/core/osgi-service-tracker-collections/src/main/java/com/liferay/osgi/service/tracker/collections/internal/list/ServiceTrackerListImpl.java:177]: @rafaprax Please fix this in a separate ticket and add test to cover it.

Thanks.

**Comment by tinatian** [comment]
@rafaprax I am going to close this pull, I believe other changes depends on the fix of service tracker list, please resend after the service tracker list bug fix is merged.

Thanks.
Tina

**Comment by rafaprax** [comment]
resent here https://github.com/liferay-core-infra/liferay-portal/pull/2613

**Pull request closed without merging**